### PR TITLE
feat: Add automatic Modrinth upload

### PR DESCRIPTION
**Pull request by zax71**
* Adds the `modrinth` Gradle task
* Modifies `release.yml` to automatically release to Modrinth

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -171,3 +172,20 @@ shadowJar {
...
+    versionNumber = version
+
+    if(System.getenv("PRERELEASE")) {
+        versionType = "beta"
+    } else {
+        versionType = "release"
+    }
+
+    uploadFile = jar
+    gameVersions = ["1.19.2", "1.19.1", "1.19", "1.18.2", "1.18.1", "1.18", "1.17.1", "1.17", "1.16.5", "1.16.4", "1.16.3", "1.16.2", "1.16.1", "1.16"]
```
Comment (benwoo1110) [inline build.gradle:188]: Since this targets the main, we should add versions 1.13-1.15

File: build.gradle [hunk 2]
```diff
@@ -171,3 +172,20 @@ shadowJar {
...
+    loaders = ["bukkit", "spigot", "paper", "purpur"]
```
Comment (benwoo1110) [inline build.gradle:189]: not very sure if we should explicitly put purpur hm

Comment (zax71) [reply build.gradle:189]: Welp, It does support it, so I don't really see why not

File: build.gradle [hunk 3]
```diff
@@ -5,6 +5,7 @@ plugins {
     id 'maven-publish'
     id 'checkstyle'
     id 'com.github.johnrengelman.shadow' version '7.1.2'
+    id "com.modrinth.minotaur" version "2.+"
```
Comment (benwoo1110) [inline build.gradle:8]: i think exact version is better

Comment (zax71) [reply build.gradle:8]: From Modrinth:
> It's safest to have this on 2.+ to get the latest features and bug fixes without having to worry about breaking changes.

Comment (benwoo1110) [reply build.gradle:8]: That is if you trust people code actually don't break stuff hehe

Comment (zax71) [reply build.gradle:8]: I think we can trust Modrinth xD

**Review by benwoo1110 — APPROVED** [review]

**Review by dumptruckman — APPROVED** [review]

**Pull request merged**